### DISCUSSION TITLE
How to configure without persistent storage

### DISCUSSION THREAD

**Discussion by adyanth** [Tzahi12345/YoutubeDL-Material#855] [Q&A] [answered]
I would like to use this as a downloader only, no persisted configs. I see a bunch of configs at `src/assets/default.json` but how do I configure them all? Can I use environment variables for all of them?

**Comment by Tzahi12345** [comment]
You can force config values but you'd still need the `default.json` config file. From what it sounds like, you're not trying to avoid "configs", rather the files downloaded to the server (like most public youtube to mp3 websites work)

This is certainly possible, I can walk you through how to set that up. In the `Extra` tab in your settings:
1. Disable `File manager enabled` (this is required for auto-delete of files)
2. Disable `Downloads manager enabled`
3. Enable `Download only mode`

The major caveat here is that the settings menu would still be accessible. There's an issue available for that (#545)

Btw, the config file you mentioned is just the dev one. The real one is in `backend/appdata`. Those are the only two config files.

**Reply by adyanth** [reply]
Thanks, what I am trying to do is run it in Kubernetes without a persistent volume. I have already set those configs (other than download manager, I need that too), and I do not need auto delete, I need them saved. How would the 3 settings persist when the pod is deleted and brought back? I can map a config json file to the pod using a configMap but I don't like to have a persistent storage for config when I do not want it to change.

**Reply by Tzahi12345** [reply, marked as answer]
Ah ok I misunderstood, yeah there's not really a way to avoid having the config.

The way it currently works is that there are environment variables (see [here](https://github.com/Tzahi12345/YoutubeDL-Material/wiki/Configuration) for a slightly outdated list) but that only overrides the settings at startup. We could definitely have an option where you can ditch the configs/settings menu and for every environment variable *not* set, it will just use the default.

Just curious on your setup, if you have a volume for your downloaded files can't you do the same for the `appdata` directory (where the config is stored)?

**Reply by adyanth** [reply]
I have an NFS mount to a location used by a media player that picks up on videos downloaded so I can watch it on my TV. I could place the config files there as well, but it would clutter the directory, nothing much else.

**Reply by Tzahi12345** [reply]
I hear you, feel free to make an issue from this so we could better keep track of it, otherwise I'll probably forget to implement it in the next version :P
